Tasks/DeployAzureResourceGroup/DeployAzureResourceGroup.ps1

```diff
@@ -9,6 +9,22 @@ param(
     [string]$sasTokenParameterName
 )
 
+function Get-SingleFile($files, $pattern)
```

Comment:
The code in this file is growing. We can move the validations, file fetching and generic functions to a utility ps1 file and refer it here. Would make it look clean